### DISCUSSION TITLE
SIGABORT while using resip in multi-threaded environment.

### DISCUSSION THREAD

**Discussion by PriyaRuhela** [resiprocate/resiprocate#374] [Q&A]
I'm encountering a SIGABRT error while using the resiprocate stack in a multi-threaded environment. The behavior seems to be random in each iteration, but it appears to be related to memory corruption around a Handle object.

Below is the core dump from the resiprocate stack:

0x00007ffff47be387 in raise () from /lib64/libc.so.6
(gdb) bt
#0  0x00007ffff47be387 in raise () from /lib64/libc.so.6
#1  0x00007ffff47bfa78 in abort () from /lib64/libc.so.6
#2  0x00007ffff47b71a6 in __assert_fail_base () from /lib64/libc.so.6
#3  0x00007ffff47b7252 in __assert_fail () from /lib64/libc.so.6
#4  0x00007ffff576853e in resip::HandleManager::getHandled (this=0x7ffff0235bb0, id=42234) at HandleManager.cxx:113
#5  0x000000000041d8b2 in resip::Handle<resip::ServerPagerMessage>::get (this=0x7fffdc87b778)
    at /home/centos/TSDK/resiprocate-1.12.6/include/resip/dum/Handle.hxx:45

Open for suggestions and any help around this.

**Comment by sgodin** [comment]
My best guess is that your application is violating the threading constraints of DUM.  Check this out: 

https://github.com/resiprocate/resiprocate/wiki/DUM-Threading
